### PR TITLE
821 data defn updates

### LINKED ISSUE #821: Data Definitions updates
# [Design](https://share.goabstract.com/9aa5ea35-8875-4f70-b05a-eb7524db8e7f)
Datarequirements

## **Meta:**
- Add Version # on the data template filename
- Add tags for Required, optional, required if (show in designs)
- Add radio buttons to choose between viewing all/only required fields
- Add unknown/not collected to all boolean fields that have that option
- Prepend "optional" to any optional field in the data template

### Update header section:
- see design for content

## **Required if fields:**
- Last funding reporting period
- Enrollment end date
- Enrollment exit reason

## **Change to optional**
- Care model
- Homelessness status
- Disability services
- Foster youth
- Birth Certificate fields (type, ID#, town of birth, state of birth)

## Data field updates
**SASID / Unique Identifier** 
- Update data definition:
  - "A unique number used to identify children. Either an SDE-backed SASID created from the PSIS system or another unique identifier from a system of record such as Childplus."

**Race**
- Add "other race" option

**Space type**
 - Replace format section with a link "See contract spaces for the full list of accepted space types." 

**First funding reporting period** 
- (Full-table width) Slim info alert with "Record 07/2020 for any child whose funding began in or before July 2020."

**Last funding reporting period** 
- required if tag + text:
"Only required for  children who: 
• Withdrew,  
• Changed age group and/or site. 
• Changed funding  source"

**Enrollment End Date and exit date:**  
- required if tag + text:
"Only required for  children who: 
• Withdrew,  
• Changed age group and/or site. 

**Enrollment end date**
  - **Update definition:**  The last date the child attended services at a site **or** the last date the child received services before changing age groups.

### PR DESCRIPTION
**Pull request by jlhogan**
closes #821
NOTE: need to land https://github.com/ctoec/data-collection/pull/836 first, and then remove the migration deletion from this changeset -- but wanted to be able to deploy to QA

todo:
- race: other option

### CODE REVIEW

**Comment by jlhogan** [comment]
@RyanHansz ready for review in QA

**Comment by RyanHansz** [comment]
Great! Lets...

- Reduce requirement levels and filter fields to h4
- Tag text - increase to 16px 
- Conditional changed to "required if"
- Reduce spacing between OEC's enrollment data requirements and Last updated: November 24, 2020 to .5rem
- Add a divider line above the data elements header to separate the sections
- Reduce spacing between "filter fields" label and radio buttons to .5 rem


**Text edits:** 
- Missing space in "You must submit your data to OECby 03/05/2021." 
- Competative to Competitive in programs
- Missing space + "created" in "A unique number used to identify children.Either an SDE-backed SASID creted from the PSIS system, or another unique identifier from a system of record such as Childplus."
- Keep race as required with "At least one race option required" below

File: client/src/components/DataDefinitionsTable/RequirementLevelFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type RequirementLevelFilterProps = {
+  setFilter: React.Dispatch<SetStateAction<boolean>>;
+};
+const TOGGLE_OPTS = { ALL_FIELDS: 'all', REQUIRED_FIELDS: 'required' };
+export const RequirementLevelFilter: React.FC<RequirementLevelFilterProps> = ({
+  setFilter,
+}) => (
+  <>
+    <div className="margin-bottom-1 margin-top-3 text-bold">Filter fields</div>
```
Comment (mmazanec22) [inline client/src/components/DataDefinitionsTable/RequirementLevelFilter.tsx:12]: We should not use fake headers.  A better alternative would be to show the legend and either use legendStyle="title" or apply a className and have a style file.

File: client/src/components/DataDefinitionsTable/RequirementLevelLegend.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { TEMPLATE_REQUIREMENT_LEVELS } from '../../shared/constants';
+import { getRequiredTag } from './utils';
+
+export const RequirementLevelLegend: React.FC = () => (
+  <>
+    <div className="margin-bottom-1 text-bold">Requirement levels</div>
```
Comment (mmazanec22) [inline client/src/components/DataDefinitionsTable/RequirementLevelLegend.tsx:7]: This should be a table caption instead

**Review by mmazanec22 — APPROVED** [review]

**Review by bamader — APPROVED** [review]

**Review by JordanGuinn — APPROVED** [review]

**Pull request merged**